Mathlib/Topology/LocallyConstant/Basic.lean

```diff
@@ -277,6 +277,14 @@ def const (X : Type*) {Y : Type*} [TopologicalSpace X] (y : Y) : LocallyConstant
 theorem coe_const (y : Y) : (const X y : X → Y) = Function.const X y :=
   rfl
 
+/-- Evaluation/projection as a locally constant function. -/
+@[simps]
+def eval {ι : Type*} {X : ι → Type*}
+    [∀ i, TopologicalSpace (X i)] [∀ i, DiscreteTopology (X i)] (i : ι) :
```

Comment:
I believe you are intentionally assuming all `X i`s to be discrete rather than just the specific needed `X i` to avoid dependent hell? If so, could you just mention it in the docstring?

Reply:
Let's just try it with only assuming `X i` being discrete. I think it will be fine.